src/git.jl

```diff
@@ -1,32 +1,56 @@
 function _git_cmd(str::AbstractString;
                   adjust_PATH::Bool = true,
                   adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path) $(split(str))`, env_mapping)
+    cmd = git(; adjust_PATH, adjust_LIBPATH)
+    return `$(cmd) $(split(str))`
 end
 
 macro git_cmd(ex)
     return _git_cmd(ex)
 end
 
-function _gitrepl_parser(repl_input::AbstractString)
-    return quote
-        GitCommand.git() do git
-            repl_input = $(Expr(:quote, repl_input))
-            run(`$(git) $(split(repl_input))`)
-            return nothing
-        end
+function git(; adjust_PATH::Bool = true, adjust_LIBPATH::Bool = true)
+    if Sys.iswindows()
+        return Git_jll.git(; adjust_PATH, adjust_LIBPATH)
     end
-end
 
-function git(;
-             adjust_PATH::Bool = true,
-             adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path)`, env_mapping...)
+    root = Git_jll.artifact_dir
+
+    libexec = joinpath(root, "libexec")
+    libexec_git_core = joinpath(libexec, "git-core")
+
+    share = joinpath(root, "share")
+    share_git_core = joinpath(share, "git-core")
+    share_git_core_templates = joinpath(share_git_core, "templates")
+
+    libpaths = [
+        dirname(Expat_jll.libexpat_path),
+        dirname(Gettext_jll.libgettext_path),
+        dirname(LibCURL_jll.libcurl_path),
+        dirname(Libiconv_jll.libiconv_path),
+        dirname(XML2_jll.libxml2_path),
+        get(ENV, JLLWrappers.LIBPATH_env, "")
+    ]
+    filter!(x -> !isempty(x), libpaths)
+    newlibpath = join(libpaths, _separator())
```

Comment:
You don't need any of this, `newlibpath` is simply `Git_jll.LIBPATH[]`.  You don't need to load all extra packages

Reply:
The `_separator` function is also not needed anymore as far as I can see, so you can delete the `utils.jl` file altogether.

Reply:
Done - I've removed those lines, and I no longer load the extra packages. I've also removed the `_separator` function and the `utils.jl` file.